react/features/base/logging/middleware.ts

```diff
@@ -150,12 +147,21 @@ function _initLogging({ dispatch, getState }: IStore,
 
     // Create the LogCollector and register it as the global log transport. It
     // is done early to capture as much logs as possible. Captured logs will be
-    // cached, before the JitsiMeetLogStorage gets ready (statistics module is
-    // initialized).
+    // cached, before the JitsiMeetLogStorage gets ready (RTCStats trace is
+    // available).
     if (!logCollector && !loggingConfig.disableLogCollector) {
-        const _logCollector = new Logger.LogCollector(new JitsiMeetLogStorage(getState));
-
-        const { apiLogLevels } = getState()['features/base/config'];
+        const { apiLogLevels, analytics: { rtcstatsLogFlushSizeBytes } = {} } = getState()['features/base/config'];
+
+        // The smaller the flush size the smaller the chance of losing logs, but
+        // the more often the logs will be sent to the server, by default the LogCollector
+        // will set once the logs reach 10KB or 30 seconds have passed since the last flush,
+        // this means if something happens between that interval and the logs don't get flushed
+        // they will be lost, for instance the meeting tab is closed, the browser crashes,
+        // an uncaught exception happens, etc.
+        // If undefined is passed the default values will be used,
+        const _logCollector = new Logger.LogCollector(new JitsiMeetLogStorage(getState), {
+            maxEntryLength: rtcstatsLogFlushSizeBytes
```

Comment:
Why is the size a parameter but not the time?

Reply:
Time would run the flush function even if there are no logs, whereas setting this to a low enough value would achieve a similar effect without running the timer when there's nothing to do.
I.e. if I set the size to 10 bytes, it would send the logs to rtcstats with every log line, which Is kind of what I plan on doing here. Where as trying to obtain the same effect (send every log line as it comes in) with a timer, would result in running the timer once every second or two seconds even if there's nothing to do.